tests/integration/transformers/test_boolean.py

```diff
@@ -85,3 +86,50 @@ def test_boolean_missing_value_generation_none(self):
         expected_reversed = pd.DataFrame({'bool': [True, True, True, False]})
         pd.testing.assert_frame_equal(transformed, expected_transformed)
         pd.testing.assert_frame_equal(reverse, expected_reversed, check_dtype=False)
+
+    def test__reverse_transform_from_manually_set_parameters_from_column(self):
+        """Test the ``reverse_transform`` after manually setting parameters."""
+        # Setup
+        data = pd.DataFrame([True, True, None, False], columns=['bool'])
+        transformed = pd.DataFrame({
+            'bool': [1.0, 1.0, 1.0, 0.0],
+            'bool.is_null': [0.0, 0.0, 1.0, 0.0],
+        })
+        column_name = 'bool'
+        transformer = BinaryEncoder()
+
+        # Run
+        null_transformer = NullTransformer('mode', missing_value_generation='from_column')
+        null_transformer._set_fitted_parameters(0.25)
+        transformer.reset_randomization()
+        transformer._set_fitted_parameters(column_name, null_transformer)
+        reverse = transformer.reverse_transform(transformed)
+
+        # Assert
+        pd.testing.assert_frame_equal(reverse, data)
+
+    def test__reverse_transform_from_manually_set_parameters_random(self):
+        """Test the ``reverse_transform`` after manually setting parameters."""
+        # Setup
+        data = pd.DataFrame([True, True, None, False, False, True, None, False], columns=['bool'])
+        transformed = pd.DataFrame({
+            'bool': [1.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0],
+        })
+        column_name = 'bool'
+        transformer = BinaryEncoder()
+
+        # Run
+        null_transformer = NullTransformer('mode', missing_value_generation='random')
+        null_transformer._set_fitted_parameters(0.2)
+        transformer._set_fitted_parameters(column_name, null_transformer)
+        reverse = transformer.reverse_transform(transformed)
+
+        # Get indices that are not NaN/None as the transformer used a random ratio
+        nan_indices_data = data[data.isna().any(axis=1)].index
+        nan_indices_reverse = reverse[reverse.isna().any(axis=1)].index
+        nan_indices = nan_indices_data.union(nan_indices_reverse)
+        compare_data = data.drop(index=nan_indices)
+        compare_reverse = data.drop(index=nan_indices)
+
+        # Assert
+        pd.testing.assert_frame_equal(compare_data, compare_reverse)
```

Comment:
We control randomness when we reverse transform so you should be able to just do this instead:
```python
expected = pd.DataFrame({'bool': [np.nan, True, np.nan, False, False, True, False, False]})
pd.testing.assert_frame_equal(expected, reverse)
```